pyanaconda/modules/payload/dnf/packages/packages.py

```diff
@@ -0,0 +1,386 @@
+
+    @property
+    def packages(self):
+        """Get list of packages marked for installation.
+
+        :rtype: [str]
+        """
+        return self._packages
+
+    def set_packages(self, packages):
+        """Set list of packages marked for installation.
+
+        :param packages: list of packages
+        :type packages: [str]
+        """
+        self._packages = packages
+        self.packages_changed.emit()
+        log.debug("Packages is set to %s.", packages)
+
+    @property
+    def excluded_groups(self):
+        """Get list of excluded groups from the installation.
+
+        :rtype: [str]
+        """
+        return self._excluded_groups
+
+    def set_excluded_groups(self, excluded_groups):
+        """Set list of excluded groups to the installation.
+
+        :param excluded_groups: list of excluded group
+        :type excluded_groups: [str]
+        """
+        self._excluded_groups = excluded_groups
+        self.excluded_groups_changed.emit()
+        log.debug("Excluded groups is set to %s.", excluded_groups)
+
+    @property
+    def excluded_packages(self):
+        """Get list of excluded packages from the installation.
+
+        :rtype: [str]
+        """
+        return self._excluded_packages
+
+    def set_excluded_packages(self, excluded_packages):
+        """Set list of excluded packages from the installation.
+
+        :param excluded_packages: list of excluded packages
+        :type excluded_packages: [str]
+        """
+        self._excluded_packages = excluded_packages
+        self.excluded_packages_changed.emit()
+        log.debug("Excluded packages is set to %s.", excluded_packages)
+
+    @property
+    def docs_excluded(self):
+        """Should the documentation be excluded during the installation.
+
+        :rtype: bool
+        """
+        return self._docs_excluded
+
+    def set_docs_excluded(self, docs_excluded):
+        """Set if the documentation should be installed with the packages.
+
+        :param docs_excluded: True if packages documentation should be installed
+        :type docs_excluded: bool
+        """
+        self._docs_excluded = docs_excluded
+        self.docs_excluded_changed.emit()
+        log.debug("Exclude docs is set to %s.", docs_excluded)
+
+    @property
+    def weakdeps_excluded(self):
+        """Should the packages weak dependencies be excluded from the installation.
+
+        :rtype: bool
+        """
+        return self._weakdeps_excluded
+
+    def set_weakdeps_excluded(self, weakdeps_excluded):
+        """Set if the weak dependencies should be excluded during the installation.
+
+        :param weakdeps_excluded: True if the weak dependencies should be excluded
+        :type weakdeps_excluded: bool
+        """
+        self._weakdeps_excluded = weakdeps_excluded
+        self.weakdeps_excluded_changed.emit()
+        log.debug("Exclude weakdeps is set to %s.", weakdeps_excluded)
+
+    @property
+    def ignore_missing(self):
```

Comment:
I would also rename the properties `ignore_missing` and `install_languages`.

Reply:
Fixed.